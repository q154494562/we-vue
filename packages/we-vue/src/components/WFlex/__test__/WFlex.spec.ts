import { mount, shallowMount } from '@vue/test-utils'
import Flex from '../'

describe('flex', () => {
  test('create', () => {
    const wrapper = mount(Flex, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-flex')
    expect(wrapper.classes()).toContain('weui-flex')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('style should be computed correctly', () => {
    const wrapper = shallowMount(Flex, {
      propsData: {
      },
    })

    expect(wrapper.vm.style).toEqual({})

    const gutterValue = 10

    wrapper.setProps({
      gutter: gutterValue,
    })

    expect(wrapper.vm.style).toEqual({
      marginLeft: `-${gutterValue / 2}px`,
      marginRight: `-${gutterValue / 2}px`,
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
