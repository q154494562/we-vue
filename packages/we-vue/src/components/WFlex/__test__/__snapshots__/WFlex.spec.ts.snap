// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flex create 1`] = `<div class="weui-flex"></div>`;

exports[`flex style should be computed correctly 1`] = `<div class="weui-flex" style="margin-left: -5px; margin-right: -5px;"></div>`;
